Wip/hughsie/drm take3

hughsie opened the pull request:
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

superm1 commented:
I added a few commits to the branch to get the parenting logic to work.  With this I can get stuff like this now:

```
├─AMD Ryzen 5 7640U w/ Radeon 760M Graphics:
│ │   Device ID:          4bde70ba4e39b28f9eab1628f9dd6e6244c03027
│ │   Current version:    0x0a704104
│ │   Vendor:             Advanced Micro Devices, Inc.
│ │   GUIDs:              b9a2dd81-159e-5537-a7db-e7101d164d3f ← cpu
│ │                       22f9ecf4-588d-5c0a-8326-6ebff3655c6d ← CPUID\PRO_0&FAM_19
│ │                       2a1df9de-fe02-59c8-8af1-32c8bb011f94 ← CPUID\PRO_0&FAM_19&MOD_74
│ │                       5a4bca32-6547-584f-bb36-920d9adede25 ← CPUID\PRO_0&FAM_19&MOD_74&STP_1
│ │   Device Flags:       • Internal device
│ │
│ ├─Graphics Processing Unit (GPU):
│ │ │   Device ID:        bc390e64ed8a9ffdab3d22dee4ed5f70edc5e319
│ │ │   Current version:  022.012.000.027.000001
│ │ │   Vendor:           Advanced Micro Devices, Inc. [AMD/ATI] (PCI:0x1002)
│ │ │   GUIDs:            74ecfd82-e883-519c-a56c-f7cd77dfc66e ← PCI\VEN_1002&DEV_15BF
│ │ │                     39fe4e5b-1660-52b2-a6e4-c5de71d57bf1 ← PCI\VEN_1002&DEV_15BF&SUBSYS_10020124
│ │ │                     6c23af4f-d6cd-5ffc-a502-0b85e472e7cb ← AMD\113-PHXGENERIC-001
│ │ │   Device Flags:     • Internal device
│ │ │
│ │ └─B140QAN02.3:
│ │       Device ID:      10feb8f8641be9b4f44d52078baa6338209a53ef
│ │       GUID:           7c2d566f-89ba-5e06-9443-fc1acd0f4929 ← DRM\VEN_AUO&DEV_E68C
│ │       Device Flags:   • Internal device
```

hughsie commented:
> I added a few commits to the branch to get the parenting logic to work

I've split these off as https://github.com/fwupd/fwupd/pull/6265 as I disagree with a few of them ;)
